cluster/aws-cloudformation/templates/kube_cluster_template.jsonnet

```diff
@@ -0,0 +1,508 @@
+          },
+        },
+        "Properties": {
+          "HealthCheckGracePeriod": "60",
+          "HealthCheckType": "EC2",
+          "LaunchConfigurationName": { "Ref": "KubernetesNodeLaunchConfig" },
+          "LoadBalancerNames": [
+            { "Ref": "KubernetesServicesELBTarget" }
+          ],
+          "MaxSize": "12",
+          "MinSize": "0",
+          "DesiredCapacity": node_count,
+          "NotificationConfigurations" :
+            alert_utils.define_auto_scaler_change_notifications_with_default_sns_alert_topic_arn(),
+          "Tags": [
+            { "Key": "Name", "Value": "kubernetes-node", "PropagateAtLaunch": "true"},
+
+            // The Kubernetes stack looks for this Tag in the EC2 metadata server as a way of isolating logically
+            // separated clusters running in the same AZ.
+            // See pkg/cloudprovider/providers/aws/aws.go#L50 in the kubernetes source for more details.
+            { "Key": "KubernetesCluster", "Value": aws_zone, "PropagateAtLaunch": "true"},
+          ],
+          "VPCZoneIdentifier": [ { "Ref": "PrivateSubnetId"} ],
+
+        }
+      },
+
+      // Setup AutoScaler alerts
+      // CPU
+      "CPUAlarmHighAlert": alert_utils.define_auto_scaler_cpu_alert_with_default_sns_topic(stack_name,
+        "KubernetesNodeAutoscaler"),
+
+      // Create an EBS volume to store long term metrics.
+      "MetricsStorageVolume": {
+        "Type": "AWS::EC2::Volume",
+        "DeletionPolicy": "Retain",  // Don't delete this if we tear down the stack.
+        "Properties": {
+          "AvailabilityZone": aws_zone,
+          "Size": "512",              // 512 GB to start with
+          "Tags": cf_utils.make_tags({ "Name": aws_zone + "-metrics-data" }),
+          "VolumeType": "gp2"
+        }
+      },
+
+      // Create an EBS volume to store logs for elasticsearch.
+      "LogsStorageVolume": {
+        "Type": "AWS::EC2::Volume",
+        "DeletionPolicy": "Retain",  // Don't delete this if we tear down the stack.
+        "Properties": {
+          "AvailabilityZone": aws_zone,
+          "Size": "1024",              // 1 TB to start with
+          "Tags": cf_utils.make_tags({ "Name": aws_zone + "-logs-data" }),
+          "VolumeType": "gp2"
+        }
+      },
+    }
+  },
+
+  etcd_endpoints(region, peer_zones)::
+    std.join(",", etcd_cluster_template.etcd_endpoints(region, peer_zones)),
+
+  absolute_kube_master_api_dns_alias(zone)::
+     "%s.kube.%s" % [zone, ops_dns_zone],
+
+   absolute_kube_services_dns_alias(zone)::
+     "%s.kube-services.%s" % [zone, ops_dns_zone],
+
+  relative_kube_master_internal_dns_alias(region, zone)::
+     dns_utils.relative_instance_dns_name(region, "kube-master-" + zone),
+
+  absolute_kube_master_internal_dns_alias(region, zone)::
+     dns_utils.absolute_instance_dns_name(region, "kube-master-" + zone),
+
+  kube_master_cloud_config(region, zone, peer_zones)::
+    importstr "kube_master_cloud_config.yaml" % {
```

Comment:
Parens are needed since v0.8.6, the reason being that people would assume that it was allowed to do importstr "foo" + ".txt" and would get confused when it would add .txt on the end of the imported file.